Lib/Coproduct.hpp

```diff
@@ -234,7 +234,7 @@ namespace CoproductImpl {
       using Ts = TL::List<As...>;                                                                             \
       auto unwrap = Unwrap<acc, Ts>{};                                                                        \
       if (acc == idx) {                                                                                       \
-        ::new (&self) TL::Get<acc,Ts>(unwrap(MOVE(value)));                                                   \
+        ::new (&self._head) TL::Get<acc,Ts>(unwrap(MOVE(value)));                                             \
```

Comment:
So I'm a little unsure what the language spec says on this, but the linter is confused and so am I. I think (same for `Option` as well) that we want to assign to one specific field of the union, not the whole thing - even though because of the memory layout that's the same thing. Did I get the right field?

Reply:
Yes absolutely! This was definitely not intended!